catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -0,0 +1,125 @@
+
+    private static final int RESOLUTION_DPI = 24;
+
+    private static final String FORMAT_NAME = "jpg";
+
+    private static final float IMAGE_QUALITY = 1.0f;
+
+    public PdfInputTransformer() {
+    }
+
+    @Override
+    public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+        return transform(input, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+        PDDocument pdfDocument = PDDocument.load(input);
+        MetacardImpl metacard = new MetacardImpl();
+
+        metacard.setId(id);
+        metacard.setMetadata("");
+
+        if (pdfDocument.isEncrypted()) {
+            LOGGER.debug("Cannot transform encrypted pdf");
+            // TODO (RCZ) - do we want to fail for any reason?
+            return metacard;
+        }
+
+        extractPdfMetadata(pdfDocument.getDocumentInformation(), metacard);
+
+        metacard.setThumbnail(generatePdfThumbnail(pdfDocument));
+
+        return metacard;
+    }
+
+    /**
+     * @param documentInformation PDF document information
+     * @param metacard A mutable metacard to add the extracted data to
+     */
+    private void extractPdfMetadata(PDDocumentInformation documentInformation,
+            MetacardImpl metacard) {
+        metacard.setContentTypeName(PDF_CONTENT_TYPE);
+
+        try {
+            Calendar creationDate = documentInformation.getCreationDate();
+            if (creationDate != null) {
+                metacard.setCreatedDate(creationDate.getTime());
+            }
+            LOGGER.info("PDF Creation date was: {}", creationDate);
+        } catch (IOException e) {
+            LOGGER.debug("Could not create date object", e);
+        }
+
+        try {
+            Calendar modificationDate = documentInformation.getModificationDate();
+            if (modificationDate != null) {
+                metacard.setCreatedDate(modificationDate.getTime());
+            }
+            LOGGER.info("PDF Modification date was: {}", modificationDate);
+
+        } catch (IOException e) {
+            LOGGER.debug("Could not create date object", e);
+        }
+
+        String title = documentInformation.getTitle();
+        if (StringUtils.isNotBlank(title)) {
+            metacard.setTitle(title);
+            // TODO (DDF-1887) - remove hardcoded setting when groomer plugin is updated
+           metacard.setMetadata("<pdf><title>" + title + "</title></pdf>");
+        }
+    }
+
+    @Override
+    public byte[] generatePdfThumbnail(InputStream pdfInputStream) throws IOException {
+        PDDocument pdfDocument = PDDocument.load(pdfInputStream);
+        if (pdfDocument.isEncrypted()) {
+            throw new IOException("Unable to read encrypted document");
+        }
+        return generatePdfThumbnail(pdfDocument);
+    }
+
+    private byte[] generatePdfThumbnail(PDDocument pdfDocument) throws IOException {
```

Comment:
What is the size of the generated thumbnail?  The metacard interface says the thumbnail attribute should be < 128K.